src/java.net.http/share/classes/java/net/http/HttpRequest.java

```diff
@@ -366,6 +379,7 @@ public static Builder newBuilder(HttpRequest request, BiPredicate<String, String
                     switch (method) {
                         case "GET" -> builder.GET();
                         case "DELETE" -> builder.DELETE();
+                        case "HEAD" -> builder.HEAD();
```

Comment:
The comment above (line 376)  mentions:
```
                // otherwise, the body is absent, special case for GET/DELETE,
                // or else use empty body
```
maybe HEAD should be added there as well...

Reply:
You're right. I missed that. I've updated the PR to fix this.